docs/toolhive/guides-k8s/connect-clients.mdx

```diff
@@ -0,0 +1,1092 @@
+    # Traefik example: strip path prefix
+    traefik.ingress.kubernetes.io/router.middlewares: toolhive-system-strip-mcp-prefix@kubernetescrd
+spec:
+  ingressClassName: traefik
+  tls:
+    - hosts:
+        - mcp.example.com
+      secretName: mcp-tls
+  rules:
+    - host: mcp.example.com
+      http:
+        paths:
+          # Fetch MCP server
+          - path: /fetch
+            pathType: Prefix
+            backend:
+              service:
+                name: mcp-fetch-proxy
+                port:
+                  number: 8080
+          - path: /.well-known/oauth-protected-resource/fetch/mcp
+            pathType: Exact
+            backend:
+              service:
+                name: mcp-fetch-proxy
+                port:
+                  number: 8080
+          # Another MCP server
+          - path: /<SERVER_NAME>
+            pathType: Prefix
+            backend:
+              service:
+                name: mcp-<SERVER_NAME>-proxy
+                port:
+                  number: 8080
+          - path: /.well-known/oauth-protected-resource/<SERVER_NAME>/mcp
+            pathType: Exact
+            backend:
+              service:
+                name: mcp-<SERVER_NAME>-proxy
+                port:
+                  number: 8080
+---
+# Traefik Middleware to strip path prefixes
+apiVersion: traefik.io/v1alpha1
+kind: Middleware
+metadata:
+  name: strip-mcp-prefix
+  namespace: toolhive-system
+spec:
+  stripPrefix:
+    prefixes:
+      - /fetch
+      - /<SERVER_NAME>
+```
+
+The MCP servers are accessible at `https://mcp.example.com/fetch/mcp` and
+`https://mcp.example.com/<SERVER_NAME>/mcp`.
+
+</TabItem>
+</Tabs>
+
+:::info[Service naming convention]
+
+The ToolHive operator automatically creates a Kubernetes Service for each
+MCPServer following the naming pattern `mcp-<name>-proxy`. For example, an
+MCPServer named `fetch` gets a Service named `mcp-fetch-proxy`.
+
+:::
+
+Apply the resources:
+
+```bash
+kubectl apply -f fetch-server.yaml
+kubectl apply -f fetch-ingress.yaml  # or mcp-ingress.yaml for path-based
+```
+
+Verify the Ingress is configured:
+
+```bash
+kubectl get ingress -n toolhive-system
+```
+
+### Option 2: Using Gateway API
+
+The [Gateway API](https://gateway-api.sigs.k8s.io/) is a more expressive way to
+expose services and is the successor to Ingress. This example works with Gateway
+API implementations like Cilium, Istio, Envoy Gateway, and Traefik, as well as
+cloud provider implementations like AWS Gateway API Controller, Google
+Kubernetes Engine (GKE) Gateway controller, and Azure Application Gateway for
+Containers. See the
+[full list of implementations](https://gateway-api.sigs.k8s.io/implementations/).
+
+:::tip
+
+For a complete working example using the ngrok Gateway API implementation, see
+the
+[Configure secure ingress for MCP servers on Kubernetes](../tutorials/k8s-ingress-ngrok.mdx)
+tutorial.
+
+:::
+
+**Check for an existing Gateway**
```

Comment:
Quick question about this line. Was the bold formatting intentional here or should this be an H4 heading? If it is meant to be a subsection under the Gateway API option, converting it to `#### Check for an existing Gateway` would help with document structure and make it show up in the table of contents.